lib/syntax/outline_inline.ml

```diff
@@ -0,0 +1,34 @@
+open! Prelude
+open Angstrom
+open Parsers
+
+let empty_plain _ = return (Inline.Plain "")
+
+let inline_choices config : Inline.t_with_pos Angstrom.t =
+  let p =
+    peek_char_fail >>= function
+    | '#' -> Inline.hash_tag config
+    | '[' -> Inline.nested_link_or_link config
+    | '(' -> Inline.block_reference config
+    (* | '{' -> macro config *)
+    | 'S'
+    | 'C'
+    | 'D'
+    | 's'
+    | 'c'
+    | 'd' ->
+      Inline.timestamp
+    | ' '
+    | '\t'
+    | '\n'
+    | '\r'
+    | '\012' ->
+      any_char >>= empty_plain
+    | _ -> take_till1 (fun c -> c = '#' || c = '[' || c = '(') >>= empty_plain
```

Comment:
is there a need to add 'S', 'C', ... (timestamp related peek chars)

Reply:
I think we don't need them here because timestamps are only allowed after either a heading or a drawer.
For example:
```
- heading
SCHEDULED: <2014-09-11 Thu>
```
or 
```
- heading
foo:: bar
SCHEDULED: <2014-09-11 Thu>
```

Timestamps after inline text will be ignored, for example: `some text\nSCHEDULED: <2014-09-11 Thu>`.